# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

NAME = Kiki Roeling #u372405

In [4]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 2.2.1     v purrr   0.2.4
v tibble  1.4.1     v dplyr   0.7.4
v tidyr   0.7.2     v stringr 1.2.0
v readr   1.1.1     v forcats 0.2.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [5]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3        V4        V5        V6        V7       
1 0.95478760 0.8565639 0.3097094 0.6774671 0.5383056 0.9352307 0.9923593
2 0.05511356 0.1913744 0.2651307 0.5834514 0.3651927 0.2947757 0.5259268
  V8        V9        V10         V11       V12       V13       V14      
1 0.3586756 0.7030854 0.738025896 0.7324128 0.1660934 0.6442538 0.1006624
2 0.8784199 0.4040902 0.005960232 0.2232931 0.3357765 0.9836452 0.2426583
  V15       V16       V17       V18       V19       V20       
1 0.7204941 0.3394626 0.4176258 0.1680668 0.4025627 0.38980150
2 0.7297369 0.7096132 0.4766780 0.8553515 0.4050760 0.06534276

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>%
  gather(Key, Value, V1:V20)

head(df2)

Key Value     
1 V1  0.95478760
2 V1  0.05511356
3 V2  0.85656386
4 V2  0.19137443
5 V3  0.30970937
6 V3  0.26513069

## Assignment 2

We will generate some data:

In [7]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [9]:
df5 <- full_join(df3, df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook